src/NodeVisitor/NamespaceStmtCreator.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the humbug/php-scoper package.
+ *
+ * Copyright (c) 2017 Théo FIDRY <theo.fidry@gmail.com>,
+ *                    Pádraic Brady <padraic.brady@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Humbug\PhpScoper\NodeVisitor;
+
+use Humbug\PhpScoper\NodeVisitor\Collection\NamespaceStmtCollection;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\NodeVisitorAbstract;
+
+/**
+ * Creates a namespace for whitelisted classes.
+ *
+ * ```
+ * class AppKernel {}
+ * ```
+ *
+ * =>
+ *
+ * ```
+ * namespace Humbug;
+ *
+ * class AppKernel {}
+ * ```
+ */
+final class NamespaceStmtCreator extends NodeVisitorAbstract
+{
+    private $prefix;
+
+    private $namespaceStatements;
+
+    private $globalWhitelister;
+
+    /**
+     * @param string                  $prefix
+     * @param NamespaceStmtCollection $namespaceStatements
+     * @param callable                $globalWhitelister
+     */
```

Comment:
the doc here is not useful as completely redundant with the signature, I know I may not have been consistent everywhere but I would remove it still :)